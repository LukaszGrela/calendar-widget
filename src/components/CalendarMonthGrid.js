import React from 'react';
import PropTypes from 'prop-types';
import { momentObj } from 'react-moment-proptypes';
import moment from 'moment';

const calendarDates = (now) => {
    let weeks = [
        [],
        [],
        [],
        [],
        [],
        []
    ];//6 weeks always to allow for 31days starting at last day of week

    return dates;
}

class CalendarMonthGrid extends React.Component {
    render() {
        const { now } = this.props;
        const monthLength = now.daysInMonth();
        const firstOfMonth = moment().year(now.year()).month(now.month()).date(1);
        const startsAt = firstOfMonth.day();
        console.log(startsAt, monthLength);
        return (
            <div className='month-grid'>
                {
                    // calendarDates(now).map(date => {
                    //     return <CalendarRow
                    //         className='week-header'
                    //         days={} />
                    // })
                }
                <div className='week row row-0'>
                    <div className='day day-0 spill'>29</div>
                    <div className='day day-1 spill'>30</div>
                    <div className='day day-2'>1</div>
                    <div className='day day-3'>2</div>
                    <div className='day day-4'>3</div>
                    <div className='day day-5'>4</div>
                    <div className='day day-6'>5</div>
                </div>
                <div className='week row row-1'>
                    <div className='day day-0'>6</div>
                    <div className='day day-1'>7</div>
                    <div className='day day-2'>8</div>
                    <div className='day day-3'>9</div>
                    <div className='day day-4'>10</div>
                    <div className='day day-5'>11</div>
                    <div className='day day-6'>12</div>
                </div>
                <div className='week row row-2'>
                    <div className='day day-0'>13</div>
                    <div className='day day-1'>14</div>
                    <div className='day day-2'>15</div>
                    <div className='day day-3'>16</div>
                    <div className='day day-4'>17</div>
                    <div className='day day-5'>18</div>
                    <div className='day day-6'>19</div>
                </div>
                <div className='week row row-3'>
                    <div className='day day-0'>20</div>
                    <div className='day day-1'>21</div>
                    <div className='day day-2'>22</div>
                    <div className='day day-3'>23</div>
                    <div className='day day-4'>24</div>
                    <div className='day day-5'>25</div>
                    <div className='day day-6'>26</div>
                </div>
                <div className='week row row-4'>
                    <div className='day day-0'>27</div>
                    <div className='day day-1'>28</div>
                    <div className='day day-2'>29</div>
                    <div className='day day-3'>30</div>
                    <div className='day day-4'>31</div>
                    <div className='day day-5 spill'>1</div>
                    <div className='day day-6 spill'>2</div>
                </div>
                <div className='week row row-5'>
                    <div className='day day-0 spill'>3</div>
                    <div className='day day-1 spill'>4</div>
                    <div className='day day-2 spill'>5</div>
                    <div className='day day-3 spill'>6</div>
                    <div className='day day-4 spill'>7</div>
                    <div className='day day-5 spill'>8</div>
                    <div className='day day-6 spill'>9</div>
                </div>
            </div>
        );
    }
}
CalendarMonthGrid.propTypes = {
    now: momentObj.isRequired
}

export default CalendarMonthGrid;