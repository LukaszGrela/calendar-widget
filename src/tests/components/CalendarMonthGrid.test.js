import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import CalendarMonthGrid from '../../components/CalendarMonthGrid';

let wrapper, now, date;
beforeEach(() => {
    now = moment('19790613');
    date = moment();

    wrapper = shallow(<CalendarMonthGrid
        now={now}
        date={date}
    />);
});

test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
});