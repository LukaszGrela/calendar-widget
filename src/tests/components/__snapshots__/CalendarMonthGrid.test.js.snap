// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="month-grid"
>
  <CalendarRow
    className="week row-0"
    current={"1970-01-01T00:00:00.000Z"}
    dayClicked={[Function]}
    days={
      Array [
        "1969-12-28T00:00:00.000Z",
        "1969-12-29T00:00:00.000Z",
        "1969-12-30T00:00:00.000Z",
        "1969-12-31T00:00:00.000Z",
        "1970-01-01T00:00:00.000Z",
        "1970-01-02T00:00:00.000Z",
        "1970-01-03T00:00:00.000Z",
      ]
    }
    key="0"
    now={"1979-06-12T23:00:00.000Z"}
  />
  <CalendarRow
    className="week row-1"
    current={"1970-01-01T00:00:00.000Z"}
    dayClicked={[Function]}
    days={
      Array [
        "1970-01-04T00:00:00.000Z",
        "1970-01-05T00:00:00.000Z",
        "1970-01-06T00:00:00.000Z",
        "1970-01-07T00:00:00.000Z",
        "1970-01-08T00:00:00.000Z",
        "1970-01-09T00:00:00.000Z",
        "1970-01-10T00:00:00.000Z",
      ]
    }
    key="1"
    now={"1979-06-12T23:00:00.000Z"}
  />
  <CalendarRow
    className="week row-2"
    current={"1970-01-01T00:00:00.000Z"}
    dayClicked={[Function]}
    days={
      Array [
        "1970-01-11T00:00:00.000Z",
        "1970-01-12T00:00:00.000Z",
        "1970-01-13T00:00:00.000Z",
        "1970-01-14T00:00:00.000Z",
        "1970-01-15T00:00:00.000Z",
        "1970-01-16T00:00:00.000Z",
        "1970-01-17T00:00:00.000Z",
      ]
    }
    key="2"
    now={"1979-06-12T23:00:00.000Z"}
  />
  <CalendarRow
    className="week row-3"
    current={"1970-01-01T00:00:00.000Z"}
    dayClicked={[Function]}
    days={
      Array [
        "1970-01-18T00:00:00.000Z",
        "1970-01-19T00:00:00.000Z",
        "1970-01-20T00:00:00.000Z",
        "1970-01-21T00:00:00.000Z",
        "1970-01-22T00:00:00.000Z",
        "1970-01-23T00:00:00.000Z",
        "1970-01-24T00:00:00.000Z",
      ]
    }
    key="3"
    now={"1979-06-12T23:00:00.000Z"}
  />
  <CalendarRow
    className="week row-4"
    current={"1970-01-01T00:00:00.000Z"}
    dayClicked={[Function]}
    days={
      Array [
        "1970-01-25T00:00:00.000Z",
        "1970-01-26T00:00:00.000Z",
        "1970-01-27T00:00:00.000Z",
        "1970-01-28T00:00:00.000Z",
        "1970-01-29T00:00:00.000Z",
        "1970-01-30T00:00:00.000Z",
        "1970-01-31T00:00:00.000Z",
      ]
    }
    key="4"
    now={"1979-06-12T23:00:00.000Z"}
  />
  <CalendarRow
    className="week row-5"
    current={"1970-01-01T00:00:00.000Z"}
    dayClicked={[Function]}
    days={
      Array [
        "1970-02-01T00:00:00.000Z",
        "1970-02-02T00:00:00.000Z",
        "1970-02-03T00:00:00.000Z",
        "1970-02-04T00:00:00.000Z",
        "1970-02-05T00:00:00.000Z",
        "1970-02-06T00:00:00.000Z",
        "1970-02-07T00:00:00.000Z",
      ]
    }
    key="5"
    now={"1979-06-12T23:00:00.000Z"}
  />
</div>
`;
